v2/src/log_sink_console.c

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "macro_utils/macro_utils.h"
+
+#include "c_logging/logger.h"
+#include "c_logging/log_context.h"
+
+#include "c_logging/log_sink_console.h"
+
+static void log_sink_console_log(LOG_LEVEL log_level, LOG_CONTEXT_HANDLE log_context, const char* message, const char* file, const char* func, int line)
+{
+    time_t t = time(NULL);
+    char temp[LOG_MAX_MESSAGE_LENGTH];
+
+    size_t property_value_pair_count = log_context_get_property_value_pair_count(log_context);
+    LOG_CONTEXT_PROPERTY_VALUE_PAIR* property_value_pairs = (log_context != NULL) ? log_context_get_property_value_pairs(log_context) : NULL;
+    (void)printf("%s Time:%.24s File:%s Func:%s Line:%d ", MU_ENUM_TO_STRING(LOG_LEVEL, log_level), ctime(&t), file, func, line);
+    for (size_t i = 0; i < property_value_pair_count; i++)
+    {
+        property_value_pairs[i].type->to_string(&property_value_pairs[i], temp, LOG_MAX_MESSAGE_LENGTH);
+        (void)printf("%s=%s ", property_value_pairs[i].name, temp);
```

Comment:
>printf [](http://example.com/codeflow?start=14&length=6)

looks like printing these values from different threads leaves the console in a "haywire" condition. I believe there should be just 1 printf / log that contains everything.

Reply:
Will do my best when implementing the console sink.
I guess we could always barf it in the temp that we anyway have here.